router/router.go

```diff
@@ -255,13 +256,13 @@ func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *R
 		videoEndpoint = aspects.QueuedRequestTimeout(videoEndpoint, cfg.RequestTimeoutHeaders, r.MetricsEngine, metrics.ReqTypeVideo)
 	}
 
-	r.POST("/openrtb2/auction", openrtbEndpoint)
-	r.POST("/openrtb2/video", videoEndpoint)
-	r.GET("/openrtb2/amp", ampEndpoint)
-	r.GET("/info/bidders", infoEndpoints.NewBiddersEndpoint(cfg.BidderInfos))
-	r.GET("/info/bidders/:bidderName", infoEndpoints.NewBiddersDetailEndpoint(cfg.BidderInfos))
-	r.GET("/bidders/params", NewJsonDirectoryServer(schemaDirectory, paramsValidator))
-	r.POST("/cookie_sync", endpoints.NewCookieSyncEndpoint(syncersByBidder, cfg, gdprPermsBuilder, tcf2CfgBuilder, r.MetricsEngine, analyticsRunner, accounts, activeBidders).Handle)
+	r.POST("/openrtb2/auction", gzipMiddleware(openrtbEndpoint))
```

Comment:
Hey… can we move this middleware wrap to here https://github.com/prebid/prebid-server/blob/master/main.go#L80-L81 so that it's on everything rather than playing whack-a-mole with every endpoint?

It can be conditionally wrapped based on the configuration and removed from or refactored out of https://github.com/prebid/prebid-server/blob/master/server/server.go#L96 - in fact, yeah, you can just make that getCompressionEnabledHandler public and use that to wrap the entire router at the top level main…

Reply:
by "just making public" you may also wish to move to some central util somewhere if it doesn't make sense in that module any longer…

Reply:
For better or worse, we started using https://github.com/CAFxX/httpcompression in our fork - because… Chrome started sending Accept-Encoding with zstd - which we hear is about the same to compress (cpu and size) but about 10x faster to decompress, if I understand correctly (may just be rumor) but, in any case, this library can handle all 4 that chrome sends (gzip, br, deflate, zstd).